modules/apps/frontend-token/frontend-token-definition-impl/src/main/java/com/liferay/frontend/token/definition/internal/TokenDefinitionImpl.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.token.definition.internal;
+
+import com.liferay.frontend.token.definition.TokenDefinition;
+
+/**
+ * @author Iván Zaera
+ */
+public class TokenDefinitionImpl implements TokenDefinition {
+
+	public TokenDefinitionImpl(String rawTokenDefinition, String themeId) {
```

Comment:
Is this necessary? `themeId`?

Maybe we could just call this "id" or "bundleId" or something else so it's just an overloaded way to map a tokenDefinitino to a key?

Reply:
It is internal for now and only necessary to map specific token definitions (those coming from themes => for now all of them => for now the one and only => `classic`) to theme ids. 

But is not exported to the outside world and will probably be removed/refactored once we have token definitions for some other type of bundle (not themes).

Reply:
But `public String getThemeId()` is not... won't that be a problem?

Reply:
You're saying no, because we just don't export this, right?

Reply:
I need themeId to manage the two maps (one per bundle, another per theme) because I only track bundles, not themes, to update the maps. So, when a bundle is removed, I need to delete the token definition from the per-theme map too. Other than that, it shouldn't go there, as I explained in the PR description.

This is a bit of hack due to the phase we are in. In the future, when we have token definitions in other type of bundles we may need more methods like `getThemeTokenDefinition` receiving other parameters (say a `someOtherThingId` instead of a `themeId`) so we will need more maps and probably a different way to track them.

But until we get to there, why making this more complex, I guess...

Reply:
Another possibility is that, instead of making more methods like `getThemeTokenDefinition` we move them to higher levels. But again, until we have the use case clear, I don't think we need to make any decision regarding that...

Reply:
Fair enough